src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -179,6 +179,7 @@ public ReadOnlySequence<T> Slice(long start, SequencePosition end)
         {
             if (start < 0)
                 ThrowHelper.ThrowStartOrEndArgumentValidationException(start);
+            BoundsCheck(_sequenceStart, end);
```

Comment:
Were these additional bounds checks necessary to make the new tests pass?

Reply:
It throw exception if `end` is before `_sequenceStart`.